homeassistant/components/hydrawise/sensor.py

```diff
@@ -130,28 +174,29 @@ async def async_setup_entry(
     entities: list[HydrawiseSensor] = []
     for controller in coordinator.data.controllers.values():
         entities.extend(
-            HydrawiseSensor(coordinator, description, controller, zone_id=zone.id)
-            for zone in controller.zones
-            for description in ZONE_SENSORS
+            HydrawiseSensor(coordinator, description, controller)
+            for description in CONTROLLER_SENSORS
         )
         entities.extend(
-            HydrawiseSensor(coordinator, description, controller, sensor_id=sensor.id)
-            for sensor in controller.sensors
+            HydrawiseSensor(coordinator, description, controller)
             for description in FLOW_CONTROLLER_SENSORS
-            if "flow meter" in sensor.model.name.lower()
+            if coordinator.data.daily_water_summary[controller.id].total_use is not None
```

Comment:
Pull this into a standalone if-statement and then conditionally add the FLOW_CONTROLLER_SENSORS and ZONE_SENSORS. Since the conditional isn't something related to what you're iterating over, it's weird having it in the list comprehension.

Reply:
Good call. Fixed.